app/models/exercise.rb

```diff
@@ -58,8 +58,8 @@ def self.for(track_slug, exercise_slug)
     self.git_important_files_hash = Git::GenerateHashForImportantExerciseFiles.(self) if self.git_important_files_hash.blank?
   end
 
-  before_update do
-    self.git_important_files_hash = Git::GenerateHashForImportantExerciseFiles.(self) if git_sha_changed?
+  after_update do
+    Exercise::UpdateHashForImportantExerciseFiles.(self)
```

Comment:
I had to change this to `after_update` as `UpdateHashForImportantExerciseFiles` itself calls `.update` on the exercise.